vscode-dotnet-runtime-library/src/Acquisition/DotnetResolver.ts

```diff
@@ -82,6 +95,117 @@ export class DotnetResolver implements IDotnetResolver
+        }
+        else
+        {
+            return this.getHostArchitectureViaInfo(dotnetExecutablePath, requestedArchitecture);
+        }
+    }
+
+    /**
+     *
+     * @param tentativePath The path to resolve.
+     * @param suggestedArchitecture The architecture to resolve the paths for - this is only a suggestion and you must validate afterward
+     * @returns The resolved true path or null if not found.
+     */
+    public async resolveTruePath(tentativePath: string, suggestedArchitecture: string | null): Promise<string | null>
+    {
+        const truePaths = await this.resolveTruePaths([tentativePath], suggestedArchitecture);
+        return truePaths?.[0] ?? null;
+    }
+
+    /**
+    *
+    * @param tentativePaths Paths that may hold a dotnet executable.
+    * @param suggestedArchitecture The architecture to resolve the paths for - this is only a suggestion and you must validate afterward
+    * @returns The actual physical location/path on disk where the executables lie for each of the paths.
+    * Some of the symlinks etc resolve to a path which works but is still not the actual path.
+    */
+    public async resolveTruePaths(tentativePaths: string[], suggestedArchitecture: string | null): Promise<string[]>
+    {
+        const oldLookup = process.env.DOTNET_MULTILEVEL_LOOKUP;
+        // This is deprecated but still needed to scan .NET 6 and below
+        process.env.DOTNET_MULTILEVEL_LOOKUP = '0'; // make it so --list-runtimes only finds the runtimes on that path: https://learn.microsoft.com/en-us/dotnet/core/compatibility/deployment/7.0/multilevel-lookup#reason-for-change
+
+        try
+        {
+            const truePaths = [];
+            suggestedArchitecture ??= DotnetCoreAcquisitionWorker.defaultArchitecture()
+
+            for (const tentativePath of tentativePaths)
+            {
+                // This will even work if only the sdk is installed, list-runtimes on an sdk installed host would work
+                const runtimeInfo = await this.getRuntimes(tentativePath, suggestedArchitecture);
+                if ((runtimeInfo?.length ?? 0) > 0)
+                {
+                    // The .NET install layout is a well known structure on all platforms.
+                    // See https://github.com/dotnet/designs/blob/main/accepted/2020/install-locations.md#net-core-install-layout
+                    //
+                    // Therefore we know that the runtime path is always in <install root>/shared/<runtime name>
+                    // and the dotnet executable is always at <install root>/dotnet(.exe).
+                    //
+                    // Since dotnet --list-runtimes will always use the real assembly path to output the runtime folder (no symlinks!)
+                    // we know the dotnet executable will be two folders up in the install root.
+                    const truePath = path.join(path.dirname(path.dirname(runtimeInfo[0].directory)), getDotnetExecutable());
```

Comment:
Do you care if the path exists, or is this just to determine what the expected path should be?